src/edu/washington/escience/myria/expression/BinaryExpression.java

```diff
@@ -89,15 +89,25 @@ protected final String getInfixBinaryString(
    */
   protected final String getObjectComparisonString(
       final SimplePredicate.Op op, final ExpressionOperatorParameter parameters) {
-    return new StringBuilder("(")
-        .append(getLeft().getJavaString(parameters))
-        .append(".compareTo(")
-        .append(getRight().getJavaString(parameters))
-        .append(')')
-        .append(op.toJavaString())
-        .append(0)
-        .append(")")
-        .toString();
+    if (op == SimplePredicate.Op.LIKE) {
```

Comment:
Also, regardless of the merits of this `LIKE` implementation, what is it doing in this PR?